java/ql/src/Security/CWE/CWE-312/SensitiveStorage.qll

```diff
@@ -243,3 +248,72 @@ class Marshallable extends ClassStore {
     )
   }
 }
+
+/* Holds if the method call is a setter method of `SharedPreferences`. */
+private predicate sharedPreferencesInput(DataFlow::Node sharedPrefs, Expr input) {
+  exists(MethodAccess m |
+    m.getMethod() instanceof SharedPreferencesSetMethod and
+    input = m.getArgument(1) and
+    sharedPrefs.asExpr() = m.getQualifier()
+  )
+}
+
+/* Holds if the method call is the store method of `SharedPreferences`. */
+private predicate sharedPreferencesStore(DataFlow::Node sharedPrefs, Expr store) {
+  exists(MethodAccess m |
+    m.getMethod() instanceof SharedPreferencesStoreMethod and
+    store = m and
+    sharedPrefs.asExpr() = m.getQualifier()
+  )
+}
+
+/* Flow from `SharedPreferences` to the method call changing its value. */
+class SharedPreferencesFlowConfig extends TaintTracking::Configuration {
+  SharedPreferencesFlowConfig() { this = "SensitiveStorage::SharedPreferencesFlowConfig" }
+
+  override predicate isSource(DataFlow::Node src) {
+    src.asExpr() instanceof SharedPreferencesEditor
+  }
+
+  override predicate isSink(DataFlow::Node sink) {
+    sharedPreferencesInput(sink, _) or
+    sharedPreferencesStore(sink, _)
+  }
+
+  override predicate isSanitizer(DataFlow::Node n) {
```

Comment:
This is a little surprising -- this is a flow of the sharedprefs object itself to one of its use sites, right? But wouldn't the encryption sanitizer relate to the argument to a sharedPreferencesInput function call, not the sharedprefs object? Perhaps illustrate a case where this sanitizer takes some action?

Reply:
Yes, the <code>isSanitizer</code> doesn't work as expected. It's not being used after I changed some other parts before submitting the first version.

I've revamped the query and added some new test cases.